docs/src/man/comparisons.md

```diff
@@ -235,6 +235,59 @@ The table below compares more advanced commands:
+|:-----------------------|:------------------------------------------------|:------------------|
+|Reduce multiple values  |`df[, list(mean(x)), by = list(id) ]`            |`combine(groupby(df, :id), :x => mean)`|
+|Add new columns (in place) |`df[, x_mean := mean(x), by = list(id) ]`        |`transform!(groupby(df, :id), :x => mean)`|
+|Pick & transform columns|`df[, list(x_mean = mean(x), y), by = list(id) ]`|`select(groupby(df, :id), :x => mean, :y)`|
+
+### More advanced commands
+
+|Operations              | data.table                                      | DataFrames.jl             |
+|:-----------------------|:------------------------------------------------|:--------------------------|
+|Complex Function          |`df[, list(mean(x, na.rm = T)) ]`              |`combine(df, :x => x -> mean(skipmissing(x)))`|
+|Transform certain rows (in place) |`df[x<=0, x:=0 ]`                      |`transform!(df, :x => (x -> ifelse.(x .<= 0, 0, x)) => :x)`|
+|Transform several columns |`df[, list(max(x), min(y)) ]`                  |`combine(df, :x => maximum,  :y => minimum)`|
+|                          |`df[, lapply(.SD, mean), .SDcols = c("x", "y") ]`   |`combine(df, [:x, :y] .=> mean)`|
+|                          |`df[, lapply(.SD, mean), .SDcols = patterns("x*") ]`|`combine(df, names(df, r"^x") .=> mean)`|
+|                          |`df[, unlist(lapply(.SD, function(x) c(max=max(x), min=min(x)))), .SDcols = c("x", "y") ]`|`combine(df, ([:x, :y] .=> [maximum minimum])...)`|
+|Multivariate function     |`df[, list(cor(x,y)) ]`                        |`transform(df, [:x, :y] => cor)`|
+|Row-wise                  |`df[, min_xy := min(x, y), by = 1:nrow(df)]`   |`transform(df, [:x, :y] => ByRow(min))`|
+|                          |`df[, argmax_xy := which.max(.SD) , .SDcols = patterns("x*"), by = 1:nrow(df) ]`|`transform(df, AsTable(r"^x") => ByRow(argmax))`|
+|DataFrame as input        |`df[, head(.SD, 2) ]`                          |`combine(d -> first(d, 2), df)`|
+|DataFrame as output       |`df[, .SD[, .(value = min(x), max(x))] ]`      |`combine(:x => x -> (value = [minimum(x), maximum(x)]), df)`|
+
+### Joining data frames
+
+| Operation             | data.table                                      | DataFrames.jl                   |
+|:----------------------|:------------------------------------------------|:--------------------------------|
+| Inner join            | `merge(df, df2, on = "grp")`                    | `innerjoin(df, df2, on = :grp)` |
+| Outer join            | `merge(df, df2, all = TRUE, on = "grp")`        | `outerjoin(df, df2, on = :grp)` |
+| Left join             | `merge(df, df2, all.x = TRUE, on = "grp")`      | `leftjoin(df, df2, on = :grp)`  |
+| Right join            | `merge(df, df2, all.y = TRUE, on = "grp")`      | `rightjoin(df, df2, on = :grp)` |
+| Anti join (filtering) | `df[!df2, on = "grp" ]`                         | `antijoin(df, df2, on = :grp)`  |
```

Comment:
probably it would be good to add then semijoin and crossjoin for completeness.

Reply:
I have not found a good standard for semi join in data.table. I can add cross join though. 
I am restricting myself to what I think is idiomatic to data.table. Perhaps there ought to be a join section within the dplyr section as well?

Reply:
OK - let us keep with what is idiomatic for data.table. This section is for people coming from other ecosystems to DataFrames.jl.

Reply:
There is one option in a data.table outer join that allows multiple match on each side with the allow.cartesian=T [flag] (https://jangorecki.gitlab.io/-/data.table/-/jobs/640724/artifacts/public/html/data.table.html).
It is more a convenience feature (merge first and filter later) than a proper merge though.

One thing that we could discuss post 1.0 is whether rolling joins is a feature that we should aim for. (I will leave this for another issue...)

Reply:
> I have not found a good standard for semi join in data.table. 

Wouldn't it just be `merge(df1, df2[, .(grp)])`? Or, wrap with with `unique()` if you want to limit to remove dups.

Reply:
I have to be honest. I don't get the point of semijoins? Is this just a filtering on the rows of df2? I have added it but it sounds a lot like more of a convenient wrapper than an actual fundamental join?

I am not a database expert though, so educate me!

Reply:
TBH I'm not much of fan either. I don't think I've ever used a semi join in any real-life project. Still, the above suggestion should get you there if you want it (not saying you do!)